src/puppetlabs/dujour/version_check.clj

```diff
@@ -130,7 +132,7 @@
   ([request-values update-server-url callback-fn]
     (validate-config! request-values update-server-url)
     (future
-      (let [key-map (and [:certname :cacert] (keys request-values))
+      (let [key-map (vec (keys (select-keys request-values [:cacert :certname])))
```

Comment:
👍 to `hashed-keys-present` or something, since these are the ones that will be hashed.